### PR TITLE
Visualization foundations

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

Extracted from https://github.com/dust-tt/dust/pull/6424 and https://github.com/dust-tt/dust/pull/6486, this PR lays the groundwork for the new visualization service. The service will render generated code in a sandbox environment, with communication facilitated via cross-document messaging. An API is implemented here allowing the visualization service to request the code and fetch files from the parent window, and it can also trigger a retry if the rendered code fails.

This component is not yet in use.

### CODE REVIEW

File: types/src/front/assistant/actions/visualization.ts [hunk 1]
```diff
@@ -25,3 +25,45 @@ export interface VisualizationActionType extends BaseAction {
...
+    extractedCode = match[1];
+  }
+  if (!extractedCode) {
+    return null;
+  }
+  return extractedCode;
+}
+
+export function visualizationExtractCodeStreaming(code: string) {
+  const startOffset = code.indexOf(">");
```
Comment (fontanierh) [inline types/src/front/assistant/actions/visualization.ts:43]: we don't wanna match the whole `<visualization>` instead ?

**Review by fontanierh — APPROVED** [review]
awesome

File: viz/tailwind.config.ts [hunk 2]
```diff
@@ -15,5 +15,10 @@ const config: Config = {
     },
   },
   plugins: [],
+  safelist: [
```
Comment (fontanierh) [inline viz/tailwind.config.ts:18]: 💡

File: types/src/front/assistant/actions/visualization.ts (hunk 1 above)
Comment (flvndvd) [reply types/src/front/assistant/actions/visualization.ts:43]: Taken from Aric's PR. I guess we will see when we use it 🙃.

**Pull request merged**